### PR TITLE
feature: S3C-1807 proxy route for metadata

### PR DESCRIPTION
**Pull request by rahulreddy**
This allows routing of requests to metadata transparently with
Cloudserver acting as a proxy enabling HTTPS, authenticated routes
for bucketd.

### CODE REVIEW

File: tests/functional/raw-node/test/routes/routeMetadata.js [hunk 1]
```diff
@@ -57,11 +57,14 @@ describe('metadata routes with metadata mock backend', () => {
         makeMetadataRequest({
             method: 'GET',
             authCredentials: metadataAuthCredentials,
-            path: '/_/metadata/listobjects/bucket1',
+            path: '/_/metadata/default/bucket/bucket1',
+            queryObj: { listingType: 'Delimiter'},
         }, (err, res) => {
+            console.log(err, res)
```
Comment (JianqinWang) [inline tests/functional/raw-node/test/routes/routeMetadata.js]: `console.log` line 63 and 67

File: tests/functional/raw-node/test/routes/routeMetadata.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ describe('metadata routes with metadata mock backend', () => {
         makeMetadataRequest({
             method: 'GET',
             authCredentials: metadataAuthCredentials,
-            path: '/_/metadata/listbuckets/1',
+            path: '/_/metadata/admin/raft_sessions/1/bucket',
```
Comment (JianqinWang) [inline tests/functional/raw-node/test/routes/routeMetadata.js:46]: https://github.com/scality/backbeat/blob/development/8.1/lib/clients/backbeat-2017-07-01.api.json
Would this file in backbeat have to be changed to reflect the new path names?

File: tests/functional/raw-node/test/routes/routeMetadata.js (hunk 1 above)
Comment (rahulreddy) [reply tests/functional/raw-node/test/routes/routeMetadata.js]: Wow Code failed me.

File: tests/functional/raw-node/test/routes/routeMetadata.js (hunk 2 above)
Comment (rahulreddy) [reply tests/functional/raw-node/test/routes/routeMetadata.js:46]: Correct, I will be making the changes. It's easier this way as we don't have to write a method for each call.

**Comment by rahulreddy** [comment]
@bert-e reset

**Comment by rahulreddy** [comment]
@bert-e reset

**Comment by rahulreddy** [comment]
@bert-e help

**Comment by rahulreddy** [comment]
@bert-e bypass_peer_approval

**Review by JianqinWang — APPROVED** [review]

**Comment by rahulreddy** [comment]
@bert-e approve

**Review by philipyoo — APPROVED** [review]

**Comment by rahulreddy** [comment]
@bert-e reset

**Comment by rahulreddy** [comment]
@bert-e reset

**Pull request merged**